nntrainer/tensor/tensor.cpp

```diff
@@ -870,6 +870,24 @@ Tensor Tensor::transpose(const std::string &direction) const {
   return result;
 }
 
+Tensor Tensor::dropout_mask(float dropout) const {
+  Tensor result(dim);
+  result.setValue(1.0);
+  Tensor rand_temp(dim);
```

Comment:
can you cache this tensor from the caller, and give the mask to this function?

Reply:
could you elaborate why?

Reply:
dropout_mask is usually stored by the dropout layer. This is because it is needed for the backwarding operation. The derivatives also needs to be zero out.

Reply:
> As far as I understand, it is not need in maks. mask is already applied in input. so.. for the training, it is already applied. But one thing I concern, the deterministic random number for the rand_temp. but nntrainer is already deteministic, so.. we don't need.

Reply:
I am concerned about the derivatives in `calcDerivative`. I think it might also need some zeroing out with the same mask.
So something like (pseudocode) `derivative *= mask`.

Reply:
> I am concerned about the derivatives in `calcDerivative`. I think it might also need some zeroing out with the same mask.
> So something like (pseudocode) `derivative *= mask`.

right. we need to save mask. I will update in later pr.